src/utils.js

```diff
@@ -455,18 +441,24 @@ export function isUnprotectedDomain (topLevelUrl, featureList) {
     return unprotectedDomain
 }
 
+/**
+ * @param {{ features: { [x: string]: { state: string; settings: any; exceptions: string[] }; }; unprotectedTemporary: string; }} data
```

Comment:
swap `[x: string]` for a `Record<string, { state: string; settings: any; exceptions: string[] }>` like below